tests/end2end-tests/data_process/test_custom_mask_data.py

```diff
@@ -0,0 +1,112 @@
+        g.edges[etype].data[name] = val
+else:
+    raise ValueError('Invalid graph format: {}'.format(args.graph_format))
+
+node1_map = read_data_parquet(os.path.join(out_dir, "raw_id_mappings", "node1"))
+reverse_node1_map = {val: key for key, val in zip(node1_map['orig'], node1_map['new'])}
+# Test the first node data
+assert g.nodes['node1'].data['feat'].dtype is th.float32
+assert g.nodes['node1'].data['feat1'].dtype is th.float32
+label = g.nodes['node1'].data['label'].numpy()
+assert label.dtype == np.int32
+data = g.nodes['node1'].data['feat'].numpy()
+data1 = g.nodes['node1'].data['feat1'].numpy()
+orig_ids = np.array([reverse_node1_map[new_id] for new_id in range(g.number_of_nodes('node1'))])
+np.testing.assert_allclose(data, orig_ids.reshape(-1, 1))
+np.testing.assert_allclose(data1, orig_ids.reshape(-1, 1))
+assert np.all(label == orig_ids % 100)
+assert 'train_mask' not in g.nodes['node1'].data
+assert 'val_mask' not in g.nodes['node1'].data
+assert 'test_mask' not in g.nodes['node1'].data
+assert np.all(np.nonzero(g.nodes['node1'].data['train_m'].numpy()) == np.arange(100))
+assert np.all(np.nonzero(g.nodes['node1'].data['val_m'].numpy()) == np.arange(100, 200))
+assert th.sum(g.nodes['node1'].data['test_m']) == 0
+
+# Test the edge data of edge type 1
+src_ids, dst_ids = g.edges(etype=('node1', 'relation1', 'node2'))
+assert 'label' in g.edges[('node1', 'relation1', 'node2')].data
+label = g.edges[('node1', 'relation1', 'node2')].data['label'].numpy()
+assert label.dtype == np.int32
+src_ids = np.array([reverse_node1_map[src_id] for src_id in src_ids.numpy()])
+dst_ids = dst_ids.numpy()
+assert np.all((src_ids + dst_ids) % 100 == label)
+assert 'train_mask' not in g.edges[('node1', 'relation1', 'node2')].data
+assert 'val_mask' not in g.nodesedges[('node1', 'relation1', 'node2')].data
+assert 'test_mask' not in g.nodesedges[('node1', 'relation1', 'node2')].data
+assert th.sum(g.edges[('node1', 'relation1', 'node2')].data['train_m']) \
+        == int(g.number_of_edges(('node1', 'relation1', 'node2')) * 0.8)
+assert th.sum(g.edges[('node1', 'relation1', 'node2')].data['val_m']) \
+        == int(g.number_of_edges(('node1', 'relation1', 'node2')) * 0.2)
+assert th.sum(g.edges[('node1', 'relation1', 'node2')].data['test_m']) == 0
+#test data type
+data = g.edges[('node1', 'relation1', 'node2')].data['feat1']
+assert data.dtype is th.float32
+data = g.edges[('node1', 'relation1', 'node2')].data['max_min_norm']
+assert data.dtype is th.float32
+assert th.max(data) <= 1.0
+assert th.min(data) >= 0
+
+# Test customized link prediction data split
+src_ids, dst_ids = g.edges(etype=('node1', 'relation_custom', 'node2'))
+src_ids = np.array([reverse_node1_map[src_id] for src_id in src_ids.numpy()])
+dst_ids = dst_ids.numpy()
+assert np.all((src_ids + dst_ids) % 100 == label)
+assert 'train_mask' not in g.edges[('node1', 'relation_custom', 'node2')].data
+assert 'val_mask' not in g.nodesedges[('node1', 'relation_custom', 'node2')].data
```

Comment:
Is this a typo? What is nodesedges here?

Reply:
typo